runtime/eoapi/raster/eoapi/raster/app.py

```diff
@@ -49,16 +66,120 @@
+                title="Tile buffer.",
+                description="Buffer on each side of the given tile. It must be a multiple of `0.5`. Output **tilesize** will be expanded to `tilesize + 2 * buffer` (e.g 0.5 = 257x257, 1.0 = 258x258).",
+            ),  # noqa
+            post_process=Depends(factory.process_dependency),  # noqa
+            rescale: Optional[List[Tuple[float, ...]]] = Depends(
+                RescalingParams
+            ),  # noqa
+            color_formula: Optional[str] = Query(
+                None,
+                title="Color Formula",
+                description="rio-color formula (info: https://github.com/mapbox/rio-color)",
+            ),  # noqa
+            colormap=Depends(factory.colormap_dependency),  # noqa
+            render_params=Depends(factory.render_dependency),  # noqa
+            pgstac_params: PgSTACParams = Depends(),  # noqa
+            backend_params=Depends(factory.backend_dependency),  # noqa
+            reader_params=Depends(factory.reader_dependency),  # noqa
+            env=Depends(factory.environment_dependency),  # noqa
+        ):
+            """Return a simple map viewer."""
+            tilejson_url = factory.url_for(
+                request, "tilejson", searchid=searchid, TileMatrixSetId=TileMatrixSetId
+            )
+
+            assets_url = factory.url_for(
+                request,
+                "assets_for_tile",
+                searchid=searchid,
+                TileMatrixSetId=TileMatrixSetId,
+                x="{x}",
+                y="{y}",
+                z="{z}",
+            )
+
+            info_url = factory.url_for(
+                request,
+                "info_search",
+                searchid=searchid,
+            )
+
+            if request.query_params._list:
+                tilejson_url += f"?{urlencode(request.query_params._list)}"
+
+            tms = factory.supported_tms.get(TileMatrixSetId)
+            return templates.TemplateResponse(
+                name="mosaic-browser.html",
+                context={
+                    "request": request,
+                    "tilejson_url": tilejson_url,
+                    "assets_url": assets_url,
+                    "info_url": info_url,
+                    "tms": tms,
+                    "resolutions": [tms._resolution(matrix) for matrix in tms],
+                },
+                media_type="text/html",
+            )
+
+
+###############################################################################
+# MOSAIC Endpoints
+# https://github.com/developmentseed/eoAPI/blob/main/runtime/eoapi/raster/eoapi/raster/app.py
 mosaic = MosaicTilerFactory(
     optional_headers=optional_headers,
     router_prefix="/mosaic",
     add_statistics=True,
-    # add /map viewer
-    add_viewer=True,
-    # add /mosaic/list endpoint
+    add_map_viewer=False,
     add_mosaic_list=True,
+    extensions=[
+        mosaicViewerExtension(),
+    ],
+    layer_dependency=AssetsBidxExprParamsOptional,
```

Comment:
why do we need to change this?

Reply:
This way you can navigate to the `/map` endpoint without specifying any parameters. Before I made this change I had to specify some parameters in the url in order to view the map (e.g. `.../map?assets=layer_name`). If I did not specify the layers I got this error message: `{"detail":"assets must be defined either via expression or assets options."}`.

Now, when you navigate to the `/map` endpoint you land in the mosaic browser web map and use the interactive menu to load specific layers rather than encoding them in the URL.

Reply:
that's pretty `dangerous` because the `layer_dependency` is also used in other endpoints. I think it will be better to not change it and use a custom parameter in the `extension` (I can do it if needed)

Reply:
Ah I see what you mean. I'll try to find an alternative way around it!

Reply:
@vincentsarago can we drop the `layer_dependency` arg from the `register()` method for the new extension? Or should we have it set to a different value there rather than inheriting from the parent `factory`?
https://github.com/ncx-co/eoAPI/blob/db40b59c2ef0b0c26dca5ebb37c9d8a51576a115/runtime/eoapi/raster/eoapi/raster/extension.py#L58

Reply:
I was going to push some change to this PR but I have couple question:
- why do we need to pass the tilejson url into the template? https://github.com/ncx-co/eoAPI/blob/map-endpoint-interactive/runtime/eoapi/raster/eoapi/raster/extension.py#L116. The tilejson URL will be created in webpage (https://github.com/ncx-co/eoAPI/blob/db40b59c2ef0b0c26dca5ebb37c9d8a51576a115/runtime/eoapi/raster/eoapi/raster/templates/mosaic-browser.html#L292-L303)

- it seems to me that we could keep the original `/map` endpoint and just add a `/viewer` endpoint with your new UI?

- we could then get rid of all of this https://github.com/ncx-co/eoAPI/blob/map-endpoint-interactive/runtime/eoapi/raster/eoapi/raster/extension.py#L46-L84

Reply:
let's use `/viewer` (it's what we use in other endpoints) 👍